lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -82,48 +80,39 @@ class IndexMapBase(ElasticBase):
     }
 
     def __init__(self, config: PbenchServerConfig, logger: Logger, *schemas: ApiSchema):
-        super().__init__(config, logger, *schemas)
-
         api_name = self.__class__.__name__
-
-        if len(schemas) != 1:
-            raise MissingDatasetNameParameter(
-                api_name, f"exactly one schema required: found {len(schemas)}"
-            )
-
-        self.schema = schemas[0]
-        dset = self.schema.get_param_by_type(
+        assert (
+            len(schemas) == 1
+        ), f"{api_name}: exactly one schema required: found {len(schemas)}"
+        dset = schemas[0].get_param_by_type(
             ParamType.DATASET,
             ApiParams(),
         )
-        if not dset or not dset.parameter.required:
-            raise MissingDatasetNameParameter(
-                api_name, "dataset parameter is not defined or not required"
-            )
-        if self.schema.authorization != ApiAuthorizationType.DATASET:
-            raise MissingDatasetNameParameter(
-                api_name, "schema authorization is not by dataset"
-            )
+        assert (
+            dset and dset.parameter.required
+        ), f"{api_name}: dataset parameter is not defined or not required"
+        assert (
+            schemas[0].authorization == ApiAuthorizationType.DATASET
+        ), f"{api_name}: schema authorization is not by dataset"
+        super().__init__(config, logger, *schemas)
+        self._method = schemas[0].method
```

Comment:
To quote someone whose judgement I respect, I find this to be too compact.  To no small extent, I'll blame Black for that (it just doesn't format two-argument assertions in a readable fashion), but when we rolled back to assertions we lost the blank lines.

I deem assertions to be "weird enough" that they should be set off from surrounding code (although not necessarily from each other...) by a blank line.  And, the call to the superclass initializer deserves similar treatment.

This will help my eye to skip over the former and to locate the latter.

Reply:
Doh!  I had this changed and then I rebase ... should be fixed now.